## pam_ldap-178-autotools.patch by Peter Marschall <peter@adpm.de>
##
## update auto* files to
## - install config & password file with the names chosen
## - make chfn and chcs use the current names of config w& password file

--- ./configure.in
+++ ./configure.in	2005-03-24 19:52:02.745066692 +0100
@@ -13,8 +13,17 @@
 AC_ARG_ENABLE(ssl, [  --disable-ssl           disable SSL/TSL support])
 AC_ARG_WITH(ldap-lib, [  --with-ldap-lib=type    select ldap library [auto|netscape5|netscape4|netscape3|umich|openldap]])
 AC_ARG_WITH(ldap-dir, [  --with-ldap-dir=DIR     base directory of ldap SDK])
-AC_ARG_WITH(ldap-conf-file, [  --with-ldap-conf-file     path to LDAP configuration file], [AC_DEFINE_UNQUOTED(PAM_LDAP_PATH_CONF, "$with_ldap_conf_file")])
-AC_ARG_WITH(ldap-secret-file, [  --with-ldap-secret-file   path to LDAP root secret file], [AC_DEFINE_UNQUOTED(PAM_LDAP_PATH_ROOTPASSWD, "$with_ldap_secret_file")])
+AC_ARG_WITH(ldap-conf-file, [  --with-ldap-conf-file     path to LDAP configuration file],
+	    [ PAM_LDAP_PATH_CONF=$with_ldap_conf_file ],
+	    [ PAM_LDAP_PATH_CONF="/etc/ldap.conf" ])
+AC_ARG_WITH(ldap-secret-file, [  --with-ldap-secret-file   path to LDAP root secret file],
+	    [ PAM_LDAP_PATH_ROOTPASSWD="$with_ldap_secret_file" ],
+	    [ PAM_LDAP_PATH_ROOTPASSWD="/etc/ldap.secret" ])
+
+AC_DEFINE_UNQUOTED(PAM_LDAP_PATH_CONF, "$PAM_LDAP_PATH_CONF")
+AC_DEFINE_UNQUOTED(PAM_LDAP_PATH_ROOTPASSWD, "$PAM_LDAP_PATH_ROOTPASSWD")
+AC_SUBST(PAM_LDAP_PATH_CONF)
+AC_SUBST(PAM_LDAP_PATH_ROOTPASSWD)
 
 if test "$ac_cv_prog_gcc" = "yes"; then CFLAGS="$CFLAGS -Wall -fPIC"; fi
 
--- ./Makefile.am
+++ ./Makefile.am	2005-03-24 19:55:05.641222603 +0100
@@ -1,7 +1,13 @@
-noinst_PROGRAMS = pam_ldap.so
+noinst_PROGRAMS = pam_ldap.so chfn chsh
 EXTRA_DIST = COPYING.LIB CVSVersionInfo.txt ChangeLog README \
 	     ldap.conf pam.conf pam_ldap.spec pam.d
 
+PAM_LDAP_PATH_CONF = @PAM_LDAP_PATH_CONF@
+PAM_LDAP_PATH_ROOTPASSWD = @PAM_LDAP_PATH_ROOTPASSWD@
+
+chfn_SOURCES = chfn.in
+chsh_SOURCES = chsh.in
+
 pam_ldap_so_SOURCES = pam_ldap.c pam_ldap.h md5.c md5.h
 pam_ldap_so_LDFLAGS = @pam_ldap_so_LDFLAGS@
 
@@ -35,12 +41,21 @@
 
 install-data-local:
 	@$(NORMAL_INSTALL)
-	@if test ! -f $(DESTDIR)$(sysconfdir)/ldap.conf; then \
-		$(mkinstalldirs) $(DESTDIR)$(sysconfdir); \
-		$(INSTALL_DATA) -o root -g root $(srcdir)/ldap.conf $(DESTDIR)$(sysconfdir)/ldap.conf; \
+	@if test ! -f $(DESTDIR)$(PAM_LDAP_PATH_CONF); then \
+		$(mkinstalldirs) $(DESTDIR)$(dir $(PAM_LDAP_PATH_CONF)); \
+		$(INSTALL_DATA) -o root -g root $(srcdir)/ldap.conf $(DESTDIR)$(PAM_LDAP_PATH_CONF); \
 	fi
 	$(INSTALL_DATA) -o root -g root $(srcdir)/pam_ldap.5 $(DESTDIR)$(mandir)/man5/pam_ldap.5
 
 uninstall-local:
 	@$(NORMAL_UNINSTALL)
 
+
+chfn$(EXEEXT):	chfn.in
+
+chsh$(EXEEXT):	chsh.in
+
+%$(EXEEXT)::	%.in
+	sed -e "s:/etc/ldap.conf:$(PAM_LDAP_PATH_CONF):" \
+	    -e "s:/etc/ldap.secret:$(PAM_LDAP_PATH_ROOTPASSWD):" < $< > $@
+
